cm/option.go

```diff
@@ -57,3 +59,25 @@ func (o option[T]) Value() T {
 	}
 	return o.some
 }
+
+// MarshalJSON implements the json.Marshaler interface.
+func (o Option[T]) MarshalJSON() ([]byte, error) {
```

Comment:
Methods should be on option[T] so named option types inherit the methods.

Reply:
Is there an example I can follow for a named option type? I assumed that if external implementations named a type it would be for `Option[T]`, perhaps I should implement the un/marshaling for both?

Reply:
@ydnar added a few more test cases that show that this works for named and nested options 🫡